dashboard/app/controllers/lti_v1_controller.rb

```diff
@@ -198,15 +197,22 @@ def sync_course
     unless Policies::Lti.roster_sync_enabled?(current_user)
       return redirect_to home_path
     end
-    params.require([:lti_integration_id, :deployment_id, :context_id, :rlid, :nrps_url]) if params[:section_code].blank?
+
+    if params[:section_code].blank?
+      begin
+        params.require([:lti_integration_id, :deployment_id, :context_id, :rlid, :nrps_url])
+      rescue ActionController::ParameterMissing => _exception
+        return render_sync_course_error("Attempting to sync a course or section from the wrong place.", :bad_request, params[:issuer], 400)
```

Comment:
Would any of these params being missing imply an invalid link placement? Or only the context_id?

Reply:
I believe just the context_id, the nrps_url will also be nil as a result of this.

Reply:
I plan to check for context_id and nrps_url here as this would mean they are launching from the wrong context then checking for deployment_id in another block. The rlid is interesting though as I believe it is an optional field. Currently both Canvas and Schoology provide this rlid but only Canvas actually makes use of it.

Reply:
Update: within this rescue I added a switch case based on the param that is missing.